### PR TITLE
Integrate with Canvas

### PR DESCRIPTION
**Pull request by wynnset**
- Add a link in iPeer to authorise iPeer with Canvas using oAuth2. 
- Once user authenticated and authorised, iPeer will retrieve the access token and refresh token, and save these values to the users table.
- Provide functionality for the user to import courses from Canvas
- When importing, for each course, the user would have the option to import the Canvas course as a new course in iPeer or link to an existing course (using a drop-down)
- When a pair of iPeer and Canvas courses are linked, the user can:
  - Import student list from Canvas into iPeer (one way only) 
    - When importing, the user can choose to either "overwrite" existing class list or "merge": If there are students enrolled in the course in iPeer that are not in Canvas, the first option will remove the users in iPeer that are not in Canvas, but the second option will keep all students.
  - Synchronise groups between Canvas and iPeer (two-way)
    - User can choose to sync from iPeer to Canvas or the other way around.
    - Since the student list between iPeer and Canvas may not be in sync, before the groups synchronisation, iPeer should retrieve the Canvas student list and compare first.  Display warning if necessary.

### CODE REVIEW

File: app/controllers/users_controller.php [hunk 1]
```diff
@@ -839,6 +839,24 @@ public function edit($userId = null, $courseId = null) {
     }
 
     /**
+     * canvasoauthreturn
+     *
+     * @access public
+     * @return void
+     */
+    function canvasoauthreturn()
```
Comment (xcompass) [inline app/controllers/users_controller.php]: Camel case for the function name?

File: app/vendors/nategood/httpful/.gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.DS_Store
```
Comment (xcompass) [inline app/vendors/nategood/httpful/.gitignore]: I prefer not to include the third party libs in the repo. Is there a specific reason why this is included. Can it be installed through composer?

Comment (kitsook) [reply app/vendors/nategood/httpful/.gitignore]: @xcompass by default, composer installs the package under the top level `vendor`.  but cakephp only looks for libraries in [the top level `vendors` folder](https://book.cakephp.org/2.0/en/core-utility-libraries/app.html#loading-vendor-files).  we could create a symbolic link in the `vendors` folder

**Comment by kitsook** [comment]
Closes #543 #544 #545

**Pull request merged**